src/map.rs

```diff
@@ -91,9 +91,10 @@ impl<T> Into<Option<T>> for RevOption<T> {
     }
 }
 
+#[derive(Debug, Clone)]
 pub enum Range<'a, K: 'a, V: 'a> {
     /// Borrowed variant.
-    Borrowed(&'a [Option<(K, V)>], usize),
+    Borrowed { slice: &'a [Option<(K, V)>], begin: usize, end: usize },
```

Comment:
These fields are public, right? Can you pack them in a private struct so that we don't expose the details?

Reply:
Indeed. I did even better: now the whole enum is private.